docs/advanced-config/README.md

```diff
@@ -71,6 +71,34 @@ end
 load(io.popen('starship init cmd'):read("*a"))()
 ```
 
+## TransientPrompt in Zsh
+
+It is possible to replace the previous-printed prompt with a custom string. This
+is useful in cases where all the prompt information is not always needed. To enable
+this, run `enable_transience` in the shell session. To make it permanent, put
+this statement in your `~/.zshrc`. Transience can be disabled on-the-fly with
+`disable_transience`.
+
+By default, the left side of input gets replaced with a bold-green `❯`. To customize this,
+define a new function called `starship_transient_prompt_func`. For example, to
+display Starship's `character` module here, you would do
+
+```shell
+function starship_transient_prompt_func {
+  starship module character
+}
+eval "$(starship init zsh)"
+enable_transience
+```
+
+Note: The current zsh transient prompt implementation has some known limitations:
+
+- The shell behaves as if `notify` option was always off (note that it's on by default).
+- The exit status from the shell on Ctrl-D is aways 0.
+
+Refer to the related [issue](https://github.com/starship/starship/issues/888) and
+[pull request](https://github.com/starship/starship/pull/4205) for more details.
```

Comment:
Let's put it on the top as an actual warning block:
````suggestion
::: warning

The current zsh transient prompt implementation has some known limitations:

- The shell behaves as if `notify` option was always off (note that it's on by default).
- The exit status from the shell on Ctrl-D is aways 0.

Refer to the related [issue](https://github.com/starship/starship/issues/888) and
[pull request](https://github.com/starship/starship/pull/4205) for more details.

:::

It is possible to replace the previous-printed prompt with a custom string. This
is useful in cases where all the prompt information is not always needed. To enable
this, run `enable_transience` in the shell session. To make it permanent, put
this statement in your `~/.zshrc`. Transience can be disabled on-the-fly with
`disable_transience`.

By default, the left side of input gets replaced with a bold-green `❯`. To customize this,
define a new function called `starship_transient_prompt_func`. For example, to
display Starship's `character` module here, you would do

```shell
function starship_transient_prompt_func {
  starship module character
}
eval "$(starship init zsh)"
enable_transience
```
````